src/rpc/mining.cpp

```diff
@@ -891,14 +893,27 @@ UniValue getblocksubsidy(const UniValue& params, bool fHelp)
     if (nHeight < 0)
         throw JSONRPCError(RPC_INVALID_PARAMETER, "Block height out of range");
 
-    CAmount nReward = GetBlockSubsidy(nHeight, Params().GetConsensus());
+    auto consensus = Params().GetConsensus();
+    CAmount nBlockSubsidy = GetBlockSubsidy(nHeight, consensus);
+    CAmount nMinerReward = nBlockSubsidy;
     CAmount nFoundersReward = 0;
-    if ((nHeight > 0) && (nHeight <= Params().GetConsensus().GetLastFoundersRewardBlockHeight(nHeight))) {
-        nFoundersReward = nReward/5;
-        nReward -= nFoundersReward;
-    }
+    bool canopyActive = consensus.NetworkUpgradeActive(nHeight, Consensus::UPGRADE_CANOPY);
+
     UniValue result(UniValue::VOBJ);
-    result.pushKV("miner", ValueFromAmount(nReward));
+    if (canopyActive) {
+        UniValue fundingstreams(UniValue::VOBJ);
+        auto fsinfos = Consensus::GetActiveFundingStreams(nHeight, consensus);
+        for (auto fsinfo : fsinfos) {
+            CAmount nStreamAmount = fsinfo.Value(nBlockSubsidy);
+            fundingstreams.pushKV(fsinfo.recipient, ValueFromAmount(nStreamAmount));
+            nMinerReward -= nStreamAmount;
+        }
+        result.pushKV("fundingstreams", fundingstreams);
```

Comment:
This is currently structured as an object with one key per funding stream:
```
"fundingstreams": {
    "Recipient A": ZEC_VALUE_A,
    "Recipient B": ZEC_VALUE_B,
}
```
However, the recipient name is intended as an informational field, not a funding stream identifier; the latter would at a minimum have to mention ZIP 214.

I would prefer this to be structured as a list of objects:
```
"fundingstreams": [
    {
        "recipient": "Recipient A",
        "value": ZEC_VALUE_A
    },
    {
        "recipient": "Recipient B",
        "value": ZEC_VALUE_B
    }
]
```
This way, we don't need an explicit FS identifier, we don't have to worry about the recipient names changing, and we can extend this API with additional information later on.

Reply:
Ok, will change. I'll also add a "specification" field since that's easy with this layout.

Reply:
```
$ src/zcash-cli getblocksubsidy 1046400
{
  "fundingstreams": [
    {
      "recipient": "Electric Coin Company",
      "specification": "https://zips.z.cash/zip-0214",
      "value": 0.21875000
    },
    {
      "recipient": "Zcash Foundation",
      "specification": "https://zips.z.cash/zip-0214",
      "value": 0.15625000
    },
    {
      "recipient": "Major Grants",
      "specification": "https://zips.z.cash/zip-0214",
      "value": 0.25000000
    }
  ],
  "miner": 2.50000000,
  "founders": 0.00000000
}
```